src/osd/PrimaryLogPG.cc

```diff
@@ -10078,6 +10078,70 @@ void PrimaryLogPG::cancel_copy_ops(bool requeue, vector<ceph_tid_t> *tids)
   }
 }
 
+struct C_gather : public Context {
+  PrimaryLogPGRef pg;
+  hobject_t oid;
+  PrimaryLogPG::OpContext *ctx;
```

Comment:
How does this get cleaned up if we don't run execute_ctx?

Reply:
I think the answer is that it gets cleared in cancel_cls_gather as the same pointer is captured in the cls_gather_ops map.  Please remove this member here and access if from cls_gather_ops -- you already have to do the lookup below.